GPIO test fails to compile for rzt2m_starter_kit

**Describe the bug**
`tests/drivers/gpio/gpio_api_1pin/drivers.gpio.1pin` fails to compile for `rzt2m_starter_kit`. It compiles with warning which is treated as error by twister.

```
FAILED: zephyr/drivers/gpio/CMakeFiles/drivers__gpio.dir/gpio_rzt2m.c.obj 
ccache /opt/toolchains/zephyr-sdk-0.16.8/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc -DKERNEL -DK_HEAP_MEM_POOL_SIZE=0 -DPICOLIBC_DOUBLE_PRINTF_SCANF -DTC_RUNID=1ad5047eeaa9fe5e7f8c7985aa85c9cd -D__LINUX_ERRNO_EXTENSIONS__ -D__PROGRAM_START -D__ZEPHYR_SUPERVISOR__ -D__ZEPHYR__=1 -I/__w/zephyr/zephyr/twister-out/rzt2m_starter_kit/tests/drivers/gpio/gpio_api_1pin/drivers.gpio.1pin/zephyr/include/generated/zephyr -I/__w/zephyr/zephyr/include -I/__w/zephyr/zephyr/twister-out/rzt2m_starter_kit/tests/drivers/gpio/gpio_api_1pin/drivers.gpio.1pin/zephyr/include/generated -I/__w/zephyr/zephyr/soc/renesas/rzt2m -I/__w/zephyr/zephyr/soc/renesas/rzt2m/. -I/__w/zephyr/zephyr/subsys/testsuite/include -I/__w/zephyr/zephyr/subsys/testsuite/coverage -I/__w/zephyr/zephyr/subsys/testsuite/ztest/include -I/__w/zephyr/modules/hal/cmsis/CMSIS/Core_R/Include -I/__w/zephyr/zephyr/modules/cmsis/. -isystem /__w/zephyr/zephyr/lib/libc/common/include -Wshadow -fno-strict-aliasing -Werror -Os -imacros /__w/zephyr/zephyr/twister-out/rzt2m_starter_kit/tests/drivers/gpio/gpio_api_1pin/drivers.gpio.1pin/zephyr/include/generated/zephyr/autoconf.h -fno-common -g -gdwarf-4 -fdiagnostics-color=always -mcpu=cortex-r52+nofp.dp -mabi=aapcs -mfpu=fpv5-sp-d16 -mfloat-abi=hard -mfp16-format=ieee -mtp=soft --sysroot=/opt/toolchains/zephyr-sdk-0.16.8/arm-zephyr-eabi/arm-zephyr-eabi -imacros /__w/zephyr/zephyr/include/zephyr/toolchain/zephyr_stdint.h -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wdouble-promotion -Wno-pointer-sign -Wpointer-arith -Wexpansion-to-defined -Wno-unused-but-set-variable -Werror=implicit-int -fno-pic -fno-pie -fno-asynchronous-unwind-tables -ftls-model=local-exec -fno-reorder-functions --param=min-pagesize=0 -fno-defer-pop -fmacro-prefix-map=/__w/zephyr/zephyr/tests/drivers/gpio/gpio_api_1pin=CMAKE_SOURCE_DIR -fmacro-prefix-map=/__w/zephyr/zephyr=ZEPHYR_BASE -fmacro-prefix-map=/__w/zephyr=WEST_TOPDIR -ffunction-sections -fdata-sections --specs=picolibc.specs -std=c99 -MD -MT zephyr/drivers/gpio/CMakeFiles/drivers__gpio.dir/gpio_rzt2m.c.obj -MF zephyr/drivers/gpio/CMakeFiles/drivers__gpio.dir/gpio_rzt2m.c.obj.d -o zephyr/drivers/gpio/CMakeFiles/drivers__gpio.dir/gpio_rzt2m.c.obj -c /__w/zephyr/zephyr/drivers/gpio/gpio_rzt2m.c
/__w/zephyr/zephyr/drivers/gpio/gpio_rzt2m.c: In function 'rzt2m_gpio_pin_interrupt_configure':
/__w/zephyr/zephyr/drivers/gpio/gpio_rzt2m.c:354:9: error: enumeration value 'GPIO_INT_TRIG_WAKE' not handled in switch [-Werror=switch]
  354 |         switch (trig) {
      |         ^~~~~~
cc1: all warnings being treated as errors
ninja: build stopped: subcommand failed.
```

**To Reproduce**
Steps to reproduce the behavior:
1. `west build -b rzt2m_starter_kit tests/drivers/gpio/gpio_api_1pin`
2. See error

**Expected behavior**
There should be no warnings.

**Impact**
CI failures like https://github.com/zephyrproject-rtos/zephyr/pull/73913/checks?check_run_id=26167820562

**Environment (please complete the following information):**
 - Commit SHA: c37e46c9064

**Additional info**
Trigger type introduced in #66587
Driver introduced in https://github.com/zephyrproject-rtos/zephyr/commit/9265d2de0cd6f72e96e609ab0adbb9e6976bb09a and https://github.com/zephyrproject-rtos/zephyr/commit/e92c04cbebf7d4af50016eb63f7bc02088590240

--- a/drivers/gpio/gpio_rzt2m.c
+++ b/drivers/gpio/gpio_rzt2m.c
@@ -360,6 +360,9 @@ static int rzt2m_gpio_pin_interrupt_configure(const struct device *dev, gpio_pin
 		break;
 	case GPIO_INT_TRIG_BOTH:
 		md_mode = INT_BOTH_EDGE;
+		break;
+	default:
+		return -EINVAL;
 	}
 
 	rzt2m_gpio_unlock();

